sprint_1

Ilya-andreevi4 opened the pull request.

Ilya-andreevi4 committed:
sprint_1, commit_1

Ilya-andreevi4 committed:
Настроил рабочее пространство

Ilya-andreevi4 committed:
Headers + sidemenu with chats list.

Ilya-andreevi4 committed:
popper for attach media + nav + error page

Ilya-andreevi4 committed:
Проект свёрстан.

Ilya-andreevi4 committed:
Исправил ReadMe file и сделал код под один стиль

Ilya-andreevi4 committed:
Update README.md

podlovchenko commented on .parcelrc:
```diff
@@ -0,0 +1,6 @@
+  "transformers": {
+    "*.hbs": ["@anikin/parcel-transformer-handlebars"]
+  }
+}
```
В проекте в некоторых файлах есть ошибка EOF
Для git важно наличие пустой строки в конце файла
https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log

podlovchenko commented on data/user-data.ts:
```diff
@@ -0,0 +1,13 @@
+  phone: "+7 (909) 967 30 30",
+  password: "string",
+}
+export default userData;
```
Использование дефолтных экспортов, на мой взгляд, это не очень хорошая практика.

Они позволяют задавать разные имена импортируемым объектам, из-за этого страдает читабельность, экземпляры с разными именами очень трудно потом искать по проекту. Так же затрудняется рефакторинг таких импортов.

Если задавать обычный экспорт, мы можем быть почти везде точно уверены, что имена используются везде одинаковые и IDE в таком случае спокойно их сможет поменять в случае рефакторинга.

Хорошая статья на эту тему
https://humanwhocodes.com/blog/2019/01/stop-using-default-exports-javascript-module/

podlovchenko commented on main-page-add-user.hbs:
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <title>Add User</title>
+    <link rel="stylesheet" href="style/index.pcss"/>
+    <script type="module" href="src/components/modal/modal.js"></script>
+  </head>
```
Эту часть можно выделить в layout

podlovchenko commented on package.json:
```diff
@@ -0,0 +1,41 @@
+    "crypto-browserify": "3.12.0",
+    "eslint": "8.33.0",
+    "events": "3.3.0",
+    "express": "4.18.2",
```
В devDependencies указывается то, что необходимо для сборки и разработки приложения, в dependencies то, что необходимо непосредственно для работу приложения (express)

podlovchenko commented on package.json:
```diff
@@ -0,0 +1,41 @@
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "parcel build index.hbs --no-cache",
+    "start": "parcel index.hbs --no-cache --port 3000"
```
Команда start должна запускать сервер на express

podlovchenko commented on package.json:
```diff
@@ -0,0 +1,41 @@
+  "author": "Ilya Orekhov",
+  "license": "ISC",
+  "devDependencies": {
+    "@anikin/parcel-transformer-handlebars": "1.0.9",
+    "@anikin/parcel-transformer-handlebars-precompile": "1.0.1",
+    "@typescript-eslint/eslint-plugin": "5.50.0",
+    "@typescript-eslint/parser": "5.50.0",
+    "assert": "2.0.0",
+    "autoprefixer": "10.4.13",
+    "browserify-zlib": "0.2.0",
+    "buffer": "5.7.1",
+    "crypto-browserify": "3.12.0",
+    "eslint": "8.33.0",
+    "events": "3.3.0",
+    "express": "4.18.2",
+    "parcel": "2.8.3",
+    "path-browserify": "1.0.1",
+    "postcss": "8.4.21",
+    "process": "0.11.10",
+    "punycode": "1.4.1",
+    "querystring-es3": "0.2.1",
+    "stream-browserify": "3.0.0",
+    "stream-http": "3.2.0",
+    "string_decoder": "1.3.0",
+    "typescript": "4.9.5",
+    "url": "0.11.0",
+    "util": "0.12.5"
```
Точно ли нужны все эти зависимости?

podlovchenko commented on server.js:
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
+
+const app = express();
+const PORT = 3000;
```
Часто требуется возможность получить порт снаружи скрипта, не изменяя код, это можно сделать через переменное окружение:
process.env.PORT || PORT;
https://nodejs.org/dist/latest-v8.x/docs/api/process.html

podlovchenko commented on src/components/friends-info/friends-info.pcss:
```diff
@@ -0,0 +1,53 @@
+  height: 24px;
+  font-weight: bold;
+  border-radius: 50px;
+  background-color: #FF7F05;
```
Цвета лучше указывать через css переменные, так будет понятнее, что это за цвет при просмотре кода и проще редактировать цвета по всему проекту, придерживаться одной цветовой палитры

podlovchenko commented on src/components/nav/nav.pcss:
```diff
@@ -0,0 +1,49 @@
+  background: rgba(0, 0, 0, 0.4);
+}
+
+.nav a {
```
Лучше использовать классы, а не теги
Статья на эту тему https://frontstuff.io/you-need-to-stop-targeting-tags-in-css

podlovchenko reviewed (changes requested):
Привет, спасибо за проделанную работу, оставила некоторые замечания
По заданию команда start должна запускать сервер на express

Ilya-andreevi4 committed:
modified:   .gitignore
	modified:   .nvmrc
	modified:   .parcelrc
	modified:   data/friends-data.ts
	new file:   data/page-list.ts
	modified:   data/user-data.ts
	modified:   handlebars.config.js
	deleted:    index.hbs
	new file:   interfaces/IFriends.ts
	new file:   interfaces/IUser.ts
	deleted:    main-page-add-user.hbs
	deleted:    main-page-empty.hbs
	modified:   package-lock.json
	modified:   package.json
	modified:   server.js
	deleted:    src/components/nav/nav.hbs
	deleted:    src/components/pop-list-item/pop-list-item.hbs
	deleted:    src/pages/entry-page/entry-page.hbs
	deleted:    src/pages/entry-page/registration-page.hbs
	deleted:    src/pages/error-page/error-page.hbs
	deleted:    src/pages/profile-page/profile-page.hbs
	deleted:    src/pages/profile-page/profile-page_change-avatar.hbs
	deleted:    src/pages/profile-page/profile-page_change-avatar_complieted.hbs
	deleted:    src/pages/profile-page/profile-page_change-avatar_error.hbs
	deleted:    src/pages/profile-page/profile-page_change-password.hbs
	deleted:    src/pages/profile-page/profile-page_change-profile.hbs
	renamed:    src/components/avatar/avatar.pcss -> static/styles/avatar.css
	renamed:    src/components/button/button.pcss -> static/styles/button.css
	renamed:    src/pages/main-page/chat-menu/chat-header/chat-header.pcss -> static/styles/chat-header.css
	renamed:    src/pages/main-page/chat-menu/chat-input/chat-input.pcss -> static/styles/chat-input.css
	renamed:    src/pages/main-page/chat-menu/chat-main/chat-main.pcss -> static/styles/chat-main.css
	renamed:    src/pages/main-page/chat-menu/chat-menu.pcss -> static/styles/chat-menu.css
	renamed:    src/pages/entry-page/entry-page.pcss -> static/styles/entry-page.css
	renamed:    src/pages/error-page/error-page.pcss -> static/styles/error-page.css
	renamed:    src/pages/main-page/side-menu/friends-container/friends-container.pcss -> static/styles/friends-container.css
	renamed:    src/components/friends-info/friends-info.pcss -> static/styles/friends-info.css
	renamed:    src/components/icon/icon.pcss -> static/styles/icon.css
	renamed:    src/components/message/message.pcss -> static/styles/message.css
	renamed:    src/components/modal/modal.pcss -> static/styles/modal.css
	renamed:    src/components/nav/nav.pcss -> static/styles/nav.css
	renamed:    src/components/pop-list-item/pop-list-item.pcss -> static/styles/pop-list-item.css
	renamed:    src/pages/main-page/side-menu/profile-header/profile-header.pcss -> static/styles/profile-header.css
	renamed:    src/pages/profile-page/profile-page.pcss -> static/styles/profile-page.css
	renamed:    src/pages/main-page/side-menu/side-menu.pcss -> static/styles/side-menu.css
	new file:   static/styles/style.css
	deleted:    style/index.pcss
	new file:   views/entry-page.hbs
	new file:   views/error-page.hbs
	new file:   views/index.hbs
	new file:   views/layouts/mainLayout.handlebars
	renamed:    src/components/modal/modal.js -> views/layouts/modal.js
	new file:   views/main-page-add-user.hbs
	new file:   views/main-page-empty.hbs
	new file:   views/media/Mail1.svg
	new file:   views/media/arrow-back.svg
	new file:   views/media/arrow-down.png
	new file:   views/media/arrow-down.svg
	new file:   views/media/attachment.svg
	new file:   views/media/avatar-1.jpg
	new file:   views/media/avatar-2.jpg
	new file:   views/media/avatar-3.jpg
	new file:   views/media/avatar-4.jpg
	new file:   views/media/avatar-5.jpg
	new file:   views/media/avatar-6.jpg
	new file:   views/media/avatar-box.png
	new file:   views/media/chat-photo.jpg
	new file:   views/media/done.svg
	new file:   views/media/file-add.svg
	new file:   views/media/group-icon.svg
	new file:   views/media/location.png
	new file:   views/media/location.svg
	new file:   views/media/media.svg
	new file:   views/media/more.svg
	new file:   views/media/search-icon.svg
	new file:   views/media/send.svg
	new file:   views/media/settings.svg
	new file:   views/media/union.svg
	new file:   views/media/user-add.svg
	renamed:    src/components/avatar/avatar.hbs -> views/partials/components/avatar/avatar.handlebars
	renamed:    src/components/button/button.hbs -> views/partials/components/button/button.handlebars
	renamed:    src/components/friends-info/friends-info.hbs -> views/partials/components/friends-info/friends-info.handlebars
	renamed:    src/components/icon/icon.hbs -> views/partials/components/icon/icon.handlebars
	renamed:    src/components/message/message.hbs -> views/partials/components/message/message.handlebars
	new file:   views/partials/components/nav/nav.handlebars
	new file:   views/partials/components/pop-list-item/pop-list-item.handlebars
	renamed:    src/pages/main-page/chat-menu/chat-header/chat-header.hbs -> views/partials/pages/main-page/chat-menu/chat-header/chat-header.handlebars
	renamed:    src/pages/main-page/chat-menu/chat-input/chat-input.hbs -> views/partials/pages/main-page/chat-menu/chat-input/chat-input.handlebars
	renamed:    src/pages/main-page/chat-menu/chat-main/chat-main.hbs -> views/partials/pages/main-page/chat-menu/chat-main/chat-main.handlebars
	renamed:    src/pages/main-page/chat-menu/chat-menu-empty.hbs -> views/partials/pages/main-page/chat-menu/chat-menu-empty.handlebars
	renamed:    src/pages/main-page/chat-menu/chat-menu.hbs -> views/partials/pages/main-page/chat-menu/chat-menu.handlebars
	renamed:    src/pages/main-page/side-menu/friends-container/friends-container.hbs -> views/partials/pages/main-page/side-menu/friends-container/friends-container.handlebars
	renamed:    src/pages/main-page/side-menu/profile-header/profile-header.hbs -> views/partials/pages/main-page/side-menu/profile-header/profile-header.handlebars
	renamed:    src/pages/main-page/side-menu/side-menu.hbs -> views/partials/pages/main-page/side-menu/side-menu.handlebars
	new file:   views/profile-page.hbs
	new file:   views/profile-page_change-avatar.hbs
	new file:   views/profile-page_change-avatar_complieted.hbs
	new file:   views/profile-page_change-avatar_error.hbs
	new file:   views/profile-page_change-password.hbs
	new file:   views/profile-page_change-profile.hbs
	new file:   views/registration-page.hbs
	new file:   views/styles/avatar.css
	new file:   views/styles/button.css
	new file:   views/styles/chat-header.css
	new file:   views/styles/chat-input.css
	new file:   views/styles/chat-main.css
	new file:   views/styles/chat-menu.css
	new file:   views/styles/entry-page.css
	new file:   views/styles/error-page.css
	new file:   views/styles/friends-container.css
	new file:   views/styles/friends-info.css
	new file:   views/styles/icon.css
	new file:   views/styles/message.css
	new file:   views/styles/modal.css
	new file:   views/styles/nav.css
	new file:   views/styles/pop-list-item.css
	new file:   views/styles/profile-header.css
	new file:   views/styles/profile-page.css
	new file:   views/styles/side-menu.css
	new file:   views/styles/style.css

Ilya-andreevi4 committed:
Add files via upload

Ilya-andreevi4 committed:
Hot fix

Ilya-andreevi4 committed:
Merge branch 'fix-branch' of into fix-branch

Ilya-andreevi4 committed:
Merge pull request #3 from Ilya-andreevi4/fix-branch

Fix branch

Ilya-andreevi4 committed:
modified:   .nvmrc
	modified:   package-lock.json
	modified:   package.json

Ilya-andreevi4 committed:
Update package.json

podlovchenko commented:
Спасибо за правки, работу принимаю)
